mobly/controllers/android_device_lib/snippet_client.py

```
@@ -74,12 +74,26 @@ def stop_app(self):
 
     def _is_app_installed(self):
         """Overrides superclass."""
+        # Check if app is installed
+        try:
+            out = self._adb.shell(
+                'pm list package | grep %s' % self.package).decode('utf-8')
+            if not bool(out):
+                return False
+        except adb.AdbError as e:
+            if (e.ret_code == 1) and (not e.stdout) and (not e.stderr):
+                return False
+            raise
+        # If app is installed, is it listed as an instrumented package.
         try:
             out = self._adb.shell(
                 'pm list instrumentation | grep ^instrumentation:%s/' %
                 self.package).decode('utf-8')
-            return bool(out)
+            is_instrumented = bool(out)
         except adb.AdbError as e:
             if (e.ret_code == 1) and (not e.stdout) and (not e.stderr):
-                return False
+                is_instrumented = False
             raise
+        if not is_instrumented:
```

That would be mixing conventions:
is_* = returns boolean
check_* = asserts something and dies if false

That's why I'm suggesting to just make check_app_installed the public-facing function. That way you can throw a better error message.